js/main.js

```diff
@@ -180,175 +191,90 @@ function viewSwap(string) {
 }
 
 function handleNav(event) {
+  if (event.target.tagName !== 'A') {
+    return;
+  }
   if (event.target.getAttribute('data-view') === 'list') {
     while ($listContainer.firstChild) {
       $listContainer.removeChild($listContainer.firstChild);
     }
     for (var i = 0; i < data.watchlist.length; i++) {
-      var $MyList = renderMyList(data.watchlist[i]);
+      var $MyList = renderWatchlist(data.watchlist[i]);
       $listContainer.appendChild($MyList);
     }
   }
   viewSwap(event.target.getAttribute('data-view'));
 }
 
-function addMovie(event) {
-  if (event.target.tagName === 'BUTTON') {
-    for (var i = 0; i < data.movies.length; i++) {
-      if (Number.parseInt(event.target.id) === data.movies[i].id) {
-        data.watchlist.unshift(data.movies[i]);
-      }
+function handleTabClick(event) {
+  if (!event.target.matches('.tab')) {
+    return;
+  }
+  for (var i = 0; i < $tabList.length; i++) {
+    if (event.target === $tabList[i]) {
+      $tabList[i].classList.add('active');
+    } else {
+      $tabList[i].classList.remove('active');
     }
   }
+  var targetData = event.target.getAttribute('data-view');
+  if (targetData === 'topRated') {
+    while ($cViewContainer.firstChild) {
+      $cViewContainer.removeChild($cViewContainer.firstChild);
+    }
+    getCategory(targetData, '', 'top_rated');
+  } else if (targetData === 'trending') {
+    while ($cViewContainer.firstChild) {
+      $cViewContainer.removeChild($cViewContainer.firstChild);
+    }
+    getCategory(targetData, 'trending/', 'week');
+  } else if (targetData === 'popular') {
+    while ($cViewContainer.firstChild) {
+      $cViewContainer.removeChild($cViewContainer.firstChild);
+    }
+    getCategory(targetData, '', 'popular');
+  } else if (targetData === 'upcoming') {
+    while ($cViewContainer.firstChild) {
+      $cViewContainer.removeChild($cViewContainer.firstChild);
+    }
+    getCategory(targetData, '', 'upcoming');
```

Comment:
This `$cViewContainer.removeChild()` is repeated four times in here. Do you always want to do it? Can you move that to before or after your `if / else` statements so it's only there once?

Reply:
Oh! You are right! Thank you!